src/client/auth/mod.rs

```diff
@@ -122,6 +122,18 @@ impl AuthMechanism {
                 };
                 Ok(())
             }
+            #[cfg(feature = "tokio-runtime")]
+            AuthMechanism::MongoDbAws => {
+                if credential.username.is_some() && credential.password.is_none() {
+                    return Err(ErrorKind::ArgumentError {
+                        message: "Username cannot be provided with password for MONGODB-AWS \
```

Comment:
I think this should be "without password"

Reply:
Oops, good catch; updated